tests/cephadm/test_cluster_view.py

```diff
@@ -0,0 +1,114 @@
+from json import loads
+
+from ceph.waiter import WaitUntil
+from cli.cephadm.cephadm import CephAdm
+from utility.log import Log
+
+log = Log(__name__)
+
+DEFAULT_CEPH_DIR = "/etc/ceph/"
+DEFAULT_CONF_PATH = "/etc/ceph/ceph.conf"
+
+
+class ClusterViewError(Exception):
+    pass
+
+
+def verify_mon_service(host):
+    """
+    Verify MON service is up
+    Args:
+        host(ceph): Installer host
+    """
+    timeout, interval = 30, 5
+    for w in WaitUntil(timeout=timeout, interval=interval):
+        configs = loads(CephAdm(host).ceph.orch.ls(service_type="mon", format="json"))
+        for config in configs:
+            running = config.get("status", {}).get("running")
+            size = config.get("status", {}).get("size")
+        if running == size:
+            log.info("All MON services are up and running")
+            return True
+        if w.expired:
+            raise ClusterViewError(
+                f"MON service failed: expected: {size}, running: {running}"
+            )
+
+
+def verify_mon_ips(host):
+    """
+    Verify MON IPs updated in ceph.conf
+    Args:
+        host(ceph): Installer host
+    """
+    # Get updated MON nodes
+    configs = loads(CephAdm(host).ceph.orch.ps(service_name="mon", format="json"))
+    mon_nodes = [config.get("daemon_id") for config in configs]
+
+    # Get MON node IPs
+    configs = loads(CephAdm(host).ceph.orch.host.ls(format="json"))
+    mon_ips = [c.get("addr") for c in configs if c.get("hostname") in mon_nodes]
+
+    # Validate MON IPs updated in ceph.conf
+    mon_ips_conf = host.remote_file(file_name=DEFAULT_CONF_PATH, file_mode="r")
+    mon_ips_conf = mon_ips_conf.readlines()[-1]
+    for mon_ip in mon_ips:
+        if mon_ip not in mon_ips_conf:
+            raise ClusterViewError("MON IPs not updated in ceph.conf")
+
+
+def run(ceph_cluster, **kwargs):
+    """Verify admin node has updated view of cluster
+    Args:
+        ceph_cluster(ceph.ceph.Ceph): CephNode or list of CephNode object
+    """
+
+    # Get installer node
+    install_node = ceph_cluster.get_nodes(role="installer")[0]
+
+    # Get non-Client nodes
+    nodes = ceph_cluster.get_nodes()
+    non_client_nodes = [c for c in nodes if "client" not in c.role]
```

Comment:
This will fail in case cluster already present on all non-client node. Add validation here to get the mon from cluster and check if expected and existing mons are not same.